File: pset6/credit/credit.py
from cs50 import get_int

# Prompt user for card number
card = get_int("Number: ")

# Assign digits to array
digits = [int(x) for x in str(card)]

# Calculate sum
sum = 0

for i in range(len(digits) - 2, -1 or -2, -2):
    if (digits[i] * 2) > 0:
        sum = sum + (int(digits[i] * 2 / 10)) % 10 + (digits[i] * 2) % 10
    else:
        sum = sum + (digits[i] * 2) % 10

for i in range(len(digits) - 1, -1 or -2, -2):
    sum = sum + digits[i]

# Check for card type
if (sum % 10) == 0:
    if len(digits) == 15 and digits[0] == 3 and (digits[1] == 4 or digits[1] == 7):
        print("AMEX")
    elif len(digits) == 16 and digits[0] == 5 and (digits[1] in [1, 2, 3, 4, 5]):
        print("MASTERCARD")
    elif (len(digits) == 13 or len(digits) == 16) and digits[0] == 4:
        print("VISA")
else:
    print("INVALID")

File: pset1/credit/credit.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>

int main(void)
{
    //get card number
    long card = get_long("Number: ");

    int digit1, digit2, product;
    int sum = 0;
    long cardnumber1 = card / 10;
    long cardnumber2 = card;

    while (cardnumber1 > 0)
    {
        digit1 = cardnumber1 % 10;

        if (digit1 * 2 >= 10)
        {
            sum = sum + (digit1 * 2) / 10 % 10 + (digit1 * 2) % 10;
        }
        else
        {
            sum = sum + (digit1 * 2) % 10;
        }
        //printf("%i\n", digit1);
        cardnumber1 = cardnumber1 / 100;
        //printf("%ld\n", cardnumber1);
    }

    while (cardnumber2 > 0)
    {
        digit2 = cardnumber2 % 10;
        sum = sum + digit2;
        cardnumber2 = cardnumber2 / 100;
        //printf("%ld\n", cardnumber2);
        //printf("%i\n", sum);
    }

    //printf("%i\n", sum);

    if ((sum % 10 == 0) && (card / 1000000000000000 == 0) && ((card / 10000000000000 == 34) || (card / 10000000000000 == 37)))
    {
        printf("AMEX\n");
    }
    else if ((sum % 10 == 0) && (card / 10000000000000000 == 0) && ((card / 100000000000000 == 51) || (card / 100000000000000 == 52)
             || (card / 100000000000000 == 53) || (card / 100000000000000 == 54) || (card / 100000000000000 == 55)))
    {
        printf("MASTERCARD\n");
    }
    else if ((sum % 10 == 0) && (((card / 10000000000000 == 0) && (card / 1000000000000 == 4)) || ((card / 10000000000000000 == 0)
                                 && (card / 1000000000000000 == 4))))
    {
        printf("VISA\n");
    }
    else
    {
        printf("INVALID\n");
    }
}